libs/cart/state/src/actions/cart-item.actions.ts

```diff
@@ -108,6 +109,10 @@ export class DaffCartItemDeleteFailure implements Action {
   constructor(public payload: string) {}
 }
 
+export class DaffCartItemStateReset implements Action {
+	readonly type = DaffCartItemActionTypes.CartItemStateResetAction;
+}
+
 export type DaffCartItemActions<
   T extends DaffCartItem = DaffCartItem,
```

Comment:
I didn't make those `DaffStatefulCartItem`s because the stateful stuff only gets added to the model in the reducers/selectors layer. The actions and effects only see `DaffCartItem`s. For that reason, I think the actions and effects should only deal with `DaffCartItem`s

Reply:
What if some effects reference some selectors? They would pass in a generic `T extends DaffCartItem` that can't satisfy the selectors generic constraint of `T extends DaffStatefulCartItem`.

Reply:
In that case, I guess we'd need to have `T extends DaffStatefulCartItem` and `V extends DaffCartItem`

Reply:
I think we should make the decision now whether or not the effects layer in general can be dependent on the selector layer. I think so, and therefore the effects layer should, in general, be dealing with the same models that the selector layer is even if it doesn't actually touch any of the fields of that model.

I'm in favor of replaciing `DaffCartItem` across the entire state package with `DaffStatefulCartItem`. I think we open ourselves up to covariance issues later on if we don't.

Reply:
done